fastestimator/schedule/lr_shedule.py

```diff
@@ -59,3 +71,125 @@ def cosine_decay(time: int,
         time_in_cycle = (time - cumulative) / current_cycle_length
         lr = (init_lr - min_lr) / 2 * math.cos(time_in_cycle * math.pi) + (init_lr + min_lr) / 2
     return lr
+
+
+@traceable()
+class ARC:
+    """A run-of-the-mill learning rate scheduler.
+
+    Args:
+        frequency: invoke frequency in terms of number of epochs.
+    """
+    def __init__(self, frequency: int = 3) -> None:
+        self.frequency = frequency
+        assert isinstance(self.frequency, int) and self.frequency > 0
+        self.model = self._load_arc_model()
+        self.lr_multiplier = {0: 1.618, 1: 1.0, 2: 0.618}
+        self.train_loss_one_cycle = []
+        self.eval_loss_one_cycle = []
+        self.all_train_loss = deque([None] * 3, maxlen=3)
+        self.all_eval_loss = deque([None] * 3, maxlen=3)
+        self.all_train_lr = deque([None] * 3, maxlen=3)
+        self.use_eval_loss = True
```

Comment:
the model doesn't support not having eval loss right? why this boolean?

Reply:
it does.    It's just not tested in experiments, but it can work.  Also, in run-of-the-mill training, we wipe away val loss 50% of the time.